Doc/library/optparse.rst

```diff
@@ -10,18 +10,47 @@
 
 **Source code:** :source:`Lib/optparse.py`
 
-.. deprecated:: 3.2
-   The :mod:`optparse` module is :term:`soft deprecated` and will not be
-   developed further; development will continue with the :mod:`argparse`
-   module.
+.. note::
+
+   :mod:`argparse` (rather than this module) is the recommended standard
+   library module for implementing command line applications unless one
+   of the following caveats applies:
+
+   * the application requires additional control over the way options and
+     positional parameters are interleaved on the command line (including
+     the ability to disable the interleaving feature completely)
+   * the application requires additional control over the incremental parsing
+     of command line elements (while ``argparse`` does support this, the
+     exact way it works in practice is undesirable for some use cases)
+   * the application requires additional control over the handling of options
+     which accept parameter values that may start with ``-`` (such as delegated
+     options to be passed to invoked subprocesses).
+
+   These ``argparse`` caveats also mean that :mod:`optparse` is likely to
+   provide a better foundation for library authors *writing* third party
+   command line argument processing libraries.
+
+.. seealso::
+
+    The :pypi:`"click" package <click>` is an ``optparse`` based third party
+    argument processing library which allows command line applications to be
+    developed as a set of appropriately decorated command implementation
+    functions.
+
+.. seealso::
+
+    The :pypi:`"Typer" package <click>` is a ``click`` based third party
+    argument processing library which allows the use of annotated Python
+    type hints to define an application's command line interface.
```

Comment:
It might make sense to duplicate these `seealso` notes in `Doc/library/argparse.rst` and `Doc/howto/argparse.rst`.

Reply:
I'm wary of the potential for confusion for folks that are running through "My first command line application" style tutorials, so I'd like to avoid having too much "Should you be using this?" text at the start of the `argparse` docs. 

Having an off-ramp from `argparse` to `optparse` and then from there to PyPI alternatives seems like it could be a reasonable middle ground.